src/couch_replicator_httpc_pool.erl

```diff
@@ -55,114 +58,113 @@ release_worker(Pool, Worker) ->
 
 init({Url, Options}) ->
     process_flag(trap_exit, true),
-    State = #state{
+    State0 = case couch_replicator_dns:lookup(Url) of
+        {ok, {Data, TTL}} ->
+            timer:send_after(TTL * 1000, refresh_dns),
+            #state{
+                dns_data = Data,
+                dns_ttl = TTL
+            };
+        {error, Error} ->
+            twig:log(
+                warn,
+                "couch_replicator_httpc_pool failed DNS lookup: ~p",
+                [Error]
+            ),
+            #state{}
+    end,
+    State = State0#state{
         url = Url,
         limit = get_value(max_connections, Options)
     },
     {ok, State}.
 
 
-handle_call(get_worker, From, State) ->
-    #state{
-        waiting = Waiting,
-        callers = Callers,
-        url = Url,
-        limit = Limit,
-        busy = Busy,
-        free = Free
-    } = State,
-    case length(Busy) >= Limit of
-    true ->
-        {noreply, State#state{waiting = queue:in(From, Waiting)}};
-    false ->
-        case Free of
-        [] ->
-           {ok, Worker} = ibrowse:spawn_link_worker_process(Url),
-           Free2 = Free;
-        [Worker | Free2] ->
-           ok
-        end,
-        NewState = State#state{
-            free = Free2,
-            busy = [Worker | Busy],
-            callers = monitor_client(Callers, Worker, From)
-        },
-        {reply, {ok, Worker}, NewState}
-    end;
+handle_call(get_worker, From, State0) ->
```

Comment:
Since this never returns a worker anymore it might be renamed queue_for_worker. I am guessing there must be source that calls get_worker somewhere that has a clause for processing a reply that is now obsolete? However, I don't know how that works with code_change.